### PR TITLE
Encode () to ObjectNil instead of ObjectArray

### PR DESCRIPTION
**Pull request by SX91**

### CODE REVIEW

File: src/Data/MessagePack/Class.hs [hunk 1]
```diff
@@ -106,8 +106,9 @@ instance MessagePack Object where
   fromObject = return
 
 instance MessagePack () where
-  toObject _ = ObjectArray []
+  toObject _ = ObjectNil
   fromObject = \case
+    ObjectNil      -> return ()
     ObjectArray [] -> return ()
```
Comment (iphydf) [inline src/Data/MessagePack/Class.hs:112]: Add `// TODO(#XXX): Remove in next version` where XXX is the number of the issue you'll create to track the removal of that code.

**Review by iphydf — CHANGES_REQUESTED** [review]

File: src/Data/MessagePack/Class.hs (hunk 1 above)
Comment (SX91) [reply src/Data/MessagePack/Class.hs:112]: Actually, I thought decoding empty arrays to () should be fine. If it's not, it's perfectly fine to remove that.

Comment (iphydf) [reply src/Data/MessagePack/Class.hs:112]: Ok, let's keep it.

**Review by iphydf — APPROVED** [review]

**Comment by iphydf** [comment]
Please rebase on master.

**Comment by SX91** [comment]
Done.

**Pull request merged**